### ISSUE TITLE
[BUG] Joint encoding of two or multiple columns outputs incomplete schema

### ISSUE THREAD

**Issue by sararb** [NVIDIA-Merlin/NVTabular#1509]
**Describe the bug**

When we jointly encode two or multiple categorical features, the int domains, and path to the categories mapping are not added to the output schema. 

**Steps/Code to reproduce bug**

This is a simple code to reproduce the schema with incomplete information:
```
import nvtabular as nvt
from merlin.io import Dataset
df = pd.DataFrame(
    {
        "Author": ["User_A", "User_E", "User_B", "User_C"],
        "Engaging User": ["User_B", "User_B", "User_A", "User_D"],
        "Post": [1, 2, 3, 4],
    }
)
cat_names = ["Post", ["Author", "Engaging User"]]
cats = cat_names >> nvt.ops.Categorify(encode_type='joint')
workflow = nvt.Workflow(cats)
df_transform = workflow.fit_transform(Dataset(df))
df_transform.schema
```
**Expected behavior**
The expected behavior is to have IndDomain property for "Author" and "Engaging User".

**Comment by karlhigley** [comment]
I think we found the cause of this while working on #1484 👍🏻

**Comment by karlhigley** [comment]
Resolved by #1511
